aioftp/client.py

```diff
@@ -603,9 +791,46 @@ class AsyncClientLister(AsyncListerMixin):
         :return: path info
         :rtype: :py:class:`dict`
         """
-        code, info = await self.command("MLST " + str(path), "2xx")
-        name, info = self.parse_mlsx_line(str.lstrip(info[1]))
-        return info
+        await self.ensure_cwd()
+
+        if type(path) is str:
+
+            path = pathlib.PurePosixPath(path)
+
+        try:
+
+            code, info = await self.command("MLST " + str(path), "2xx")
+
+        except errors.StatusCodeError as e:
+
+            if not e.received_codes[-1].matches("50x"):
+
+                return None
```

Comment:
Sometimes the file doesn't exist, and below there is that case where LIST doesn't find the file, and there is no exception to raise